### PR TITLE
Brian/make gameplay fixes and lint

### PR DESCRIPTION
**Pull request by cb299792458**

### CODE REVIEW

File: frontend/src/components/ui/BuyMenu.jsx [hunk 1]
```diff
@@ -72,6 +72,7 @@ export default function BuyMenu({game}) {
     useEffect(() => {
         // Handle game state changes
         deselectAll();
+    //eslint-disable-next-line
```
Comment (rockokw) [inline frontend/src/components/ui/BuyMenu.jsx:75]: A bit broad, could use this to make more specific:

```suggestion
    //eslint-disable-next-line react-hooks/exhaustive-deps
```


... or instead add this to `.eslintrc.cjs`:

```
'react-hooks/exhaustive-deps': 'off', // suppress useEffect dependency warnings
```

File: frontend/src/components/ui/BuyMenu.jsx [hunk 2]
```diff
@@ -155,6 +156,7 @@ export default function BuyMenu({game}) {
                 game.setPath();
             });
         }
+    //eslint-disable-next-line
```
Comment (rockokw) [inline frontend/src/components/ui/BuyMenu.jsx:159]: Same as above

```suggestion
    //eslint-disable-next-line react-hooks/exhaustive-deps
```

File: frontend/src/components/views/HpView.jsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ export default function HpView({hpFraction, position, width = 0.5, mode = "slide
             }
             materialRef.current.color = color
```
Comment (rockokw) [inline frontend/src/components/views/HpView.jsx:27]: Oh, I almost forgot about the color mode

**Review by rockokw — APPROVED** [review]
I rebased this PR so not sure what's gonna happen with our setup after you merge the first one. I guess we'll find out :grin:

**Comment by rockokw** [comment]
Cool, looks like it automatically updated the base after the other PR merged. However, the diff for that shared commit still shows changes, interesting.

File: frontend/src/components/tests/AnimatedMapDisplay.jsx [hunk 4]
```diff
@@ -116,6 +116,7 @@ const AnimatedMapDisplay = ({gameMap, tileConfig, path, enemies}) => {
         };
     
         tick();
+    // eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (cb299792458) [reply frontend/src/components/tests/AnimatedMapDisplay.jsx:119]: Yeah, let's get rid of everything but `GameDisplay`

**Pull request merged**